### PR TITLE
Update front matter, acknowledgements, criticality, and links to Defender for Teams Baseline

### PR DESCRIPTION
**Pull request by ahuynhMITRE**
## 🗣 Description ##

Updating the front-matter and acknowledgements to the AAD SCB to match language approved by the CISA Front office. Also added the html formatting for criticality to support new field in report. Also updated the links to defender now that formal review is finished. 

## 💭 Motivation and context ##

Changes are required to ensure consistency across SCBs for the approved front-matter and acknowledgements. Criticality HTML is added to support new report field and supporting automation.  Links to defender to align with policies that can be "passed" via defender. 

#471 
#341 
#435 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: baselines/teams.md [hunk 1]
```diff
@@ -568,11 +598,15 @@ Any product that meets the requirements outlined in this baseline policy group m
...
 Attachments included with Teams messages SHOULD be scanned for malware.
+
+<!--Policy: MS.TEAMS.7.1v1; Criticality: SHOULD -->
 - _Rationale:_ Teams can be used as a mechanism for delivering malware. In many cases, malware can be detected through scanning, reducing the risk for end users.
 - _Last modified:_ July 2023
 
 #### MS.TEAMS.7.2v1
 Users SHOULD be prevented from opening or downloading files detected as malware.
+
+<!--Policy: MS.TEAMS.7.1v1; Criticality: SHOULD -->
```
Comment (crutchfield) [inline baselines/teams.md:609]:
```suggestion
<!--Policy: MS.TEAMS.7.2v1; Criticality: SHOULD -->
```

**Review by crutchfield — CHANGES_REQUESTED** [review]
One minor correction. Looks good

**Review by crutchfield — APPROVED** [review]
Looks good.

**Comment by ahuynhMITRE** [comment]
@nanda-katikaneni good to merge!

**Pull request merged**